src/ContentTypeMiddleware.php

```diff
@@ -66,9 +67,9 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         }
 
         $contentType = $this->extractContentType($response->getHeaderLine('Content-Type'));
-        $formatter   = $this->formatters[$contentType] ?? null;
+        $formatter   = $this->formatters[$contentType] ?? $this->getNotAcceptableFormatter();
```

Comment:
I'm modifying this slightly to ensure that we always have the formatter when instantiating this, which removes a method call from this bit.

It's a very small performance gain for normal PHP usage but for handling things with async php is a nice thing.

Reply:
Actually I've chosen the laziest variant exactly for this small gain because this object is just not needed in normal (non error) cases.
But it is such a tiny difference that it is just unimportant

Reply:
Exactly, which is why I'm thinking if it makes sense to actually cache this instance... I mean, in a ideal scenario it will hardly reach that point.